refactor naming of CIDR-based policy

ianvernon opened the pull request:
The phrase L3 was used to refer to CIDR-based (not label-based) policy in some
parts of the codebase. This renames the CIDR-specific parts of code that refer
to CIDR based-policy as L3 policy to contain CIDR instead of L3 to make it
clearer that the relevant code only deals with CIDR-based policy, not
label-based policy.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
refactor naming of CIDR-based policy

The phrase L3 was used to refer to CIDR-based (not label-based) policy in some
parts of the codebase. This renames the CIDR-specific parts of code that refer
to CIDR based-policy as L3 policy to contain CIDR instead of L3 to make it
clearer that the relevant code only deals with CIDR-based policy, not
label-based policy.

Signed-off by: Ian Vernon <ian@cilium.io>

joestringer commented on pkg/policy/rule.go:
```diff
@@ -298,7 +298,8 @@ func (r *rule) resolveL4Policy(ctx *SearchContext, state *traceState, result *L4
 
-func mergeL3(ctx *SearchContext, dir string, ipRules []api.CIDR, ruleLabels labels.LabelArray, resMap *L3PolicyMap) int {
+// mergeL3 checks
+func mergeL3(ctx *SearchContext, dir string, ipRules []api.CIDR, ruleLabels labels.LabelArray, resMap *CIDRPolicyMap) int {
```
Rename the function to `mergeCIDR()`, perhaps?

ianvernon replied:
Done :)

joestringer reviewed (approved):
Already an improvement, thanks!

rlenglet reviewed (approved):
LGTM, thanks!

Commit:
pkg/policy: change mergeL3 to mergeCIDR

Add documentation of mergeCIDR function as well.

Signed-off by: Ian Vernon <ian@cilium.io>
